examples/sigstore-go-signing/main.go

```diff
@@ -214,22 +216,46 @@ func main() {
 		}
 	}
 
+	rekorVersion := uint32(2)
 	if *rekor {
 		rekorURLs, err := root.SelectServices(signingConfig.RekorLogURLs(),
-			signingConfig.RekorLogURLsConfig(), []uint32{1}, time.Now())
+			signingConfig.RekorLogURLsConfig(), []uint32{rekorVersion}, time.Now())
 		if err != nil {
 			log.Fatal(err)
 		}
 		for _, rekorURL := range rekorURLs {
+			var version uint32
+			for _, r := range signingConfig.RekorLogURLs() {
+				if rekorURL == r.URL {
+					version = r.MajorAPIVersion
+					break
+				}
+			}
+			var publicKey crypto.PublicKey
+			for _, t := range opts.TrustedRoot.RekorLogs() {
+				if rekorURL == t.BaseURL {
+					publicKey = t.PublicKey
+					break
+				}
+			}
+			var origin string
+			if version == 2 {
+				origin = strings.TrimPrefix(rekorURL, "https://")
+			}
 			rekorOpts := &sign.RekorOptions{
-				BaseURL: rekorURL,
-				Timeout: time.Duration(90 * time.Second),
-				Retries: 1,
+				BaseURL:   rekorURL,
+				Timeout:   time.Duration(90 * time.Second),
+				Retries:   1,
+				Version:   version,
+				Origin:    origin,
+				PublicKey: publicKey,
 			}
 			opts.TransparencyLogs = append(opts.TransparencyLogs, sign.NewRekor(rekorOpts))
 		}
 	}
 
+	opts.UseRekorV2 = rekorVersion == uint32(2)
```

Comment:
We also need to enforce that if Rekor v2 is used to sign, there is at least one timestamp authority.